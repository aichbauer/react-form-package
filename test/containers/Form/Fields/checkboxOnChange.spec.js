import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
} from '../../../../src';

configure({ adapter: new Adapter() });

test('Render Form with checkbox input | onChange', () => {
  const handleOnChangeSpy = jest.spyOn(Form.prototype, 'handleOnChange');
  const event = {
    target: {
      id: 'checkbox',
      value: true,
      checked: true,
      type: 'checkbox',
    },
  };
  const rules = {
    type: 'checkbox',
    min: -1,
    max: -1,
    required: true,
  };

  const myComponent = (
    <Form>
      <div>
        <div>checkbox</div>
        <Field type="checkbox" id="checkbox" required />
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  let state = wrapper.state();

  expect(state.data.checkbox.value).toBe(false);
  expect(state.data.checkbox.valid).toBe(false);


  wrapper.find('input').simulate('focus', event, rules);
  wrapper.find('input').simulate('change', event, rules);
  wrapper.find('input').simulate('blur', event, rules);
  wrapper.update();

  state = wrapper.state();

  expect(state.data.checkbox.value).toBe(true);
  expect(state.data.checkbox.valid).toBe(true);
  expect(handleOnChangeSpy).toHaveBeenCalled();
  expect(tree).toMatchSnapshot();
});
