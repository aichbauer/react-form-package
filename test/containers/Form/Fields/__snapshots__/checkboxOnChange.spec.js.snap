// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with checkbox input | onChange 1`] = `
<form
  className="rfp-form"
>
  <div>
    <div>
      checkbox
    </div>
    <input
      checked={false}
      className="rfp-checkbox"
      id="checkbox"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      type="checkbox"
      value={false}
    />
  </div>
</form>
`;
